sqlstream/reader.go

```diff
@@ -114,6 +117,13 @@ func (r *sqlReader) deserializeData(rawColumns [][]byte) []string {
 	return tidyCols
 }
 
+func (r *SQLReader) HasData() bool {
+	if !r.cachedNext {
+		r.cachedNext = r.rows.Next()
+	}
+	return r.cachedNext
+}
+
```

Comment:
I feel a bit hesitant about adding `HasData` as [Reader.Read](https://golang.org/pkg/io/#Reader) already satisfies the purpose that HasData is supposed to serve. I understand it lets us know beforehand, but we're just simulating "beforehand" by caching the next read.

Together with the previous comment (on correct behaviour of an empty table), perhaps we could fix the issue raised in this PR on `s5` package? Specifically by returning an error at s3Stream.UploadMultiPart saying that "you can't multipart upload empty data" and [recovering](https://github.com/omidaladini/s5/blob/master/s5/s5.go#L72) the error by PUTting an empty file. 

Basically, assigning responsibilities to respective packages.